scripts/generate

```diff
@@ -609,6 +609,27 @@ ${content}
 </html>
 `));
 
+//    program :: Array String -> Future String String
+const program = pipe ([
+  array (reject ('Missing command-line argument'))
+        (compose (K) (resolve)),
+  map (curry2 (path.join)),
+  chain (path =>
+           pair (lift3 (toDocument))
+                (bimap (chain (compose (eitherToFuture)
+                                       (field ($.String) ('version'))))
+                       (chain (compose (eitherToFuture)
+                                       (field ($.String) ('description'))))
+                       (join (Pair)
+                             (chain (compose (mapLeft (prop ('message')))
+                                             (compose (eitherToFuture)
+                                                      (encase (JSON.parse))))
+                                    (readFile (path ('package.json'))))))
+                (readme (path ('README.md')))),
```

Comment:
It took me much longer than I'd like to admit to understand this function. Maybe legibility can be improved by rewriting this segment in a left-to-right pipeline?

```js
pump ('package.json') ([
  path,
  readFile,
  chain (compose (mapLeft (prop ('message'))) (Future.encase  (JSON.parse))),
  join (Pair),
  bimap (chain (compose (eitherToFuture) (field ($.String) ('version'))))
        (chain (compose (eitherToFuture) (field ($.String) ('description')))),
  pair (lift3 (toDocument)),
  T (readme (path ('README.md'))),
])
```

(where `pump = flip (pipe)`)

Reply:
Your version reads better to me too. Updated. :)